### PR TITLE
MM-47084 : Migrate "components/code_preview.jsx" to Typescript

### PR DESCRIPTION
**Pull request by AbhinavVihan**
#### Summary
Migrated code_preview.jsx to typescript

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/21085
Fixes https://mattermost.atlassian.net/browse/MM-47084

#### Related Pull Requests
None

#### Screenshots
None

#### Release Note
NONE
```release-note
NONE
```

### CODE REVIEW

File: components/code_preview.tsx [hunk 1]
```diff
@@ -69,7 +86,7 @@ export default class CodePreview extends React.PureComponent {
         }
     }
 
-    handleReceivedCode = (data) => {
+    handleReceivedCode = (data: any) => {
```
Comment (AshishDhama) [inline components/code_preview.tsx:89]: was it not possible to figure out the type of data here, We want to avoid any as much as possible.

**Review by AshishDhama — CHANGES_REQUESTED** [review]

File: components/file_preview_modal/file_preview_modal.tsx [hunk 2]
```diff
@@ -336,7 +336,7 @@ export default class FilePreviewModal extends React.PureComponent<Props, State>
                     dialogClassName += ' modal-code';
                     content = (
                         <CodePreview
-                            fileInfo={fileInfo}
+                            fileInfo={fileInfo as FileInfo}
```
Comment (nevyangelova) [inline components/file_preview_modal/file_preview_modal.tsx:339]: does the type not match here?

Comment (nevyangelova) [inline components/file_preview_modal/file_preview_modal.tsx:339]: does the type not match here?

Comment (nevyangelova) [inline components/file_preview_modal/file_preview_modal.tsx:339]: does the type not match here?

Comment (AbhinavVihan) [reply components/file_preview_modal/file_preview_modal.tsx:339]: Yes, the type for fileInfo is **FileInfo | LinkInfo**. And without this type type mismatch error **Type 'LinkInfo' is missing the following properties from type 'FileInfo': id, user_id, create_at, update_at, and 7 more** comes up

File: components/code_preview.tsx (hunk 1 above)
Comment (AbhinavVihan) [reply components/code_preview.tsx:89]: Hi @AshishDhama, the correct type here should be **string**, But the we are accessing property **nodeName** from data which does not exist on type string. 
So neither can we give it type **Node** nor **string**

File: components/file_preview_modal/file_preview_modal.tsx (hunk 2 above)
Comment (AbhinavVihan) [reply components/file_preview_modal/file_preview_modal.tsx:339]: > Yes, the type for fileInfo is **FileInfo | LinkInfo**. And without this type, mismatch error "**Type 'LinkInfo' is missing the following properties from type 'FileInfo': id, user_id, create_at, update_at, and 7 more**" comes up

File: components/code_preview.tsx (hunk 1 above)
Comment (AbhinavVihan) [reply components/code_preview.tsx]: resolved

**Comment by nevyangelova** [comment]
@AbhinavVihan your merge with master introduced extra files. Can you rebase your fork's master with mattermost:master as well?

**Comment by AbhinavVihan** [comment]
@nevyangelova rebasing is done. Would you please take a look.

**Comment by M-ZubairAhmed** [comment]
@AbhinavVihan there are few conflicts, can you please check

**Review by nevyangelova — APPROVED** [review]

**Comment by AbhinavVihan** [comment]
hi @M-ZubairAhmed, Conflicts have been resolved. Would you please take a look.

**Review by M-ZubairAhmed — APPROVED** [review]
💯 It looks good to me. Thank you 🎉

**Review by AshishDhama — APPROVED** [review]

**Comment by AshishDhama** [comment]
@jgilliam17 it’s ready for QA

**Comment by jgilliam17** [comment]
Thanks @AbhinavVihan
Tested, looks good to merge.
[E2E report](https://mattermost-cypress-report.s3.amazonaws.com/247449-b66278c-onprem-ent-webapp-pr-11197-mm-ee-test:b66278c/mochawesome.html) - no PR related failures .

**Review by jgilliam17 — APPROVED** [review]

**Pull request merged**